hash_practice/exercises.py

```diff
@@ -2,18 +2,46 @@
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n^2)? The fucntion only loops through the length of strings for 
+                                the number of letters in each string?
+        Space Complexity: O(n)? Theres only 2 varaibles being stored...sortedWord and anagramsHash?
     """
-    pass
+    anagramsHash = {}
+    for string in strings:
+        sortedWord = "".join(sorted(string))
+        if sortedWord in anagramsHash:
+            anagramsHash[sortedWord].append(string)
+        else:
+            anagramsHash[sortedWord] = [string]
+    return list(anagramsHash.values())
+    
 
 def top_k_frequent_elements(nums, k):
     """ This method will return the k most common elements
         In the case of a tie it will select the first occuring element.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity:  O(n^2)? 

```

Comment:
Space complexity would be O(n). `finalList` will be of length `k` which can never be greater than the length of `nums`. `freqHash` will have at most n key value pairs as well, where the values are all integers.